pkg/mailboxwatch/interface.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mailboxwatch
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	restclient "k8s.io/client-go/rest"
+	upstreamcache "k8s.io/client-go/tools/cache"
+
+	kcpcache "github.com/kcp-dev/apimachinery/v2/pkg/cache"
+	kcpinformers "github.com/kcp-dev/apimachinery/v2/third_party/informers"
+	kcpscopedclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpapiinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	tenancyv1a1informers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/tenancy/v1alpha1"
+)
+
+// NewSharedInformer constructs a shared index informer on objects of a given kind in mailbox workspaces.
+// It follows the usual pattern for a constructor of informers except for an additional parameter
+// at the front, which it uses to keep appraised of the mailbox workspaces.
+// The ListerWatcher can be constructed using `upstreamcache.NewListWatchFromClient`.
+func NewSharedInformer(
+	mailboxWorkspacePreInformer tenancyv1a1informers.WorkspaceInformer,
+	listerWatcher upstreamcache.ListerWatcher,
+	exampleObject runtime.Object,
```

Comment:
Can we call this `watchedObject` or something like that ? I think it would make the function signature a bit more clear.

Reply:
But it is not _the_ watched object, the purpose here is to watch several objects, and the one passed in is not any one of them.  It is only there to convey information about the type of object to watch.  That is why "example" is on-point.